src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -63,23 +63,39 @@ internal SymbolSpecification()
 
         internal bool AppliesTo(ISymbol symbol)
         {
-            if (ApplicableSymbolKindList.Any() && !ApplicableSymbolKindList.Any(k => k.AppliesTo(symbol)))
+            return AnyMatches(this.ApplicableSymbolKindList, symbol) &&
+                   AllMatches(this.RequiredModifierList, symbol) &&
+                   AnyMatches(this.ApplicableAccessibilityList, symbol);
+        }
+
+        private bool AnyMatches<TSymbolMatcher>(ImmutableArray<TSymbolMatcher> matchers, ISymbol symbol)
+            where TSymbolMatcher : ISymbolMatcher
+        {
+            if (!matchers.Any())
```

Comment:
Looks like `Any` might throw for "default" Immutable arrays. Is that okay here (I think yes)...